fcrepo-persistence-api/src/main/java/org/fcrepo/persistence/api/PersistentStorage.java

```diff
@@ -17,10 +17,74 @@
  */
 package org.fcrepo.persistence.api;
 
+import java.time.Instant;
+
+import org.fcrepo.kernel.api.models.FedoraResource;
+
 /**
  * An interface that mediates CRUD operations to and from persistence storage.
  *
  * @author dbernstein
+ * @author whikloj
+ * @since 2019-09-11
  */
 public interface PersistentStorage {
-}
\ No newline at end of file
+
+    /**
+     * Locate an existing resource on disk.
+     *
+     * @param txID The transaction identifer or null if no transaction.
+     * @param identifier The id of the resource.
+     * @param version The datetime of the version you want to retrieve or null if current.
+     * @return the FedoraResource
+     */
+    public FedoraResource findResource(final String txID, final String identifier, final Instant version);
+
+    /**
+     * Save/Update a Resource to storage.
+     *
+     * @param txID The transaction identifier or null if no transaction.
+     * @param identifier The identifier of the resource (if known).
+     * @param resource The FedoraResource to persist.
+     * @return the identifier
+     */
+    public String saveResource(final String txID, final String identifier, final FedoraResource resource);
+
+    /**
+     * Delete an object from storage.
+     *
+     * @param txID The transaction identifer or null if no transaction.
+     * @param identifier The identifier of the resource.
+     * @param version The datetime of the version to be deleted or null for entire resource.
+     */
+    public void deleteResource(final String txID, final String identifier, final Instant version);
+
+    /**
+     * Start a new transaction.
+     *
+     * @param idleTimeout The amount of time to wait between actions in the transaction.
+     * @return the identifier of the transaction.
+     */
+    public String startTransaction(final long idleTimeout);
```

Comment:
This transaction stuff could totally be separated out to a Transaction API implementation. But to benefit from possible performance benefits they would probably need to work together to avoid replaying all the interactions again.